File: Ruby/kyu-6/_0032__Valid_Braces/solution.rb
def valid_braces(s)
  stack = []
  couples = { '(' => ')', '[' => ']', '{' =>'}' }
  startItem = '([{'
  s.each_char do |item|
    if (startItem.include?(item))
      stack.push(item)
    else
      if (stack.length === 0 || couples[stack.pop()] != item)
        return false
      end
    end

  end
  stack.size == 0;
end

File: JavaScript/kyu-6/_0032__Valid_Braces/solution.js
function validBraces(s) {
  const stack = [];
  const couples = { '(': ')', '[': ']', '{': '}' };
  const startItem = '([{';
  for (let i = 0; i < s.length; i++) {
    const item = s[i];
    if (startItem.includes(item)) {
      stack.push(item);
    } else {
      if (stack.length === 0 || couples[stack.pop()] !== item) return false;
    }
  }
  return stack.length === 0;
}
